### PR TITLE
Fix #3573: Merge section_title.xml into single xml

### LINKED ISSUE #3573: Merge section_title.xml into single xml file
Currently there are 4 versions of `section_title.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by JishnuGoyal**
## Explanation

Fixes issue #3573 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

Before (Mobile-Landscape)
image

Before (Mobile-Portrait)
image

After (Mobile-Landscape)
image

After (Mobile-Portrait)
image

Before (Tablet-Landscape)
image

Before (Tablet-Portrait)
image

After (Tablet-Landscape)
image

After (Tablet-Portrait)
image

### CODE REVIEW

**Comment by JishnuGoyal** [comment]
Kindly review my PR @rt4914

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -577,4 +577,9 @@
   <dimen name="profile_reset_pin_activity_profile_reset_input_confirm_pin_margin_end">28dp</dimen>
   <dimen name="profile_reset_pin_activity_profile_reset_save_button_margin_end">28dp</dimen>
 
+  <!-- SectionTitleFragment -->
+  <dimen name="section_title_divider_view_layout_marginTop">32dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Please rename all these variables correctly to fully camel_case format.
i.e.
`section_title_divider_view_layout_margin_top`

Similar for other files too. Thanks.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL Thanks

**Comment by rt4914** [comment]
@bkaur-bkj Can you please review this PR before I do full review?

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 2]
```diff
@@ -390,4 +390,10 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: @JishnuGoyal pls remove this extra line number 393

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 3]
```diff
@@ -392,4 +392,10 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: @JishnuGoyal same here you can remove this extra line added

File: app/src/main/res/values-land/dimens.xml [hunk 4]
```diff
@@ -377,4 +377,9 @@
   <dimen name="profile_reset_pin_activity_profile_reset_input_confirm_pin_margin_end">168dp</dimen>
   <dimen name="profile_reset_pin_activity_profile_reset_save_button_margin_end">168dp</dimen>
 
+  <!-- SectionTitleFragment -->
+  <dimen name="section_title_divider_view_layout_margin_top">28dp</dimen>
+  <dimen name="section_title_text_view_layout_margin_start">72dp</dimen>
+  <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
+  <dimen name="section_title_text_view_layout_margin_end">72dp</dimen>
```
Comment (bkaur-bkj) [inline app/src/main/res/values-land/dimens.xml:391]: @JishnuGoyal add one empty line above </ resources>  to keep it formatted like it was before edits
Similar for the other dimens file as well

Comment (JishnuGoyal) [reply app/src/main/res/values-land/dimens.xml:391]: done

**Review by bkaur-bkj — APPROVED** [review]
@JishnuGoyal please take a note that you don't need to resolve the comments by yourself because reviewers use them to verify that the requested changes are made. Other than that everything looks good to me. @rt4914 PTAL

**Review by rt4914 — CHANGES_REQUESTED** [review]
@JishnuGoyal Please remove `recently_played_tablet_port_outer_margin` and `recently_played_tablet_land_outer_margin` from all files if not required.

Thanks.

**Comment by bkaur-bkj** [comment]
@JishnuGoyal make sure you keep assignee section up to date, if the changes requested by reviewer is completed from your end, you can tag reviewer with comment as PTAL so that  bot assigns the PR to the reviewer

**Comment by JishnuGoyal** [comment]
Noted. Thank you so much for your tip, I'll keep this in mind from next time 😃 @bkaur-bkj

**Comment by JishnuGoyal** [comment]
@rt4914 PTAL

**Review by rt4914 — CHANGES_REQUESTED** [review]
@JishnuGoyal Just merge this branch with latest develop to get remove of `Merge Conflicts` and then it should be good to merge.

**Comment by JishnuGoyal** [comment]
@rt4914 PTAL

File: .idea/deploymentTargetDropDown.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+          <Key>
+            <type value="VIRTUAL_DEVICE_PATH" />
+            <value value="C:\Users\Jishnu Goyal\.android\avd\Tablet_Pixel_C_API_30.avd" />
+          </Key>
+        </deviceKey>
+      </Target>
+    </targetSelectedWithDropDown>
+    <timeTargetWasSelectedWithDropDown value="2021-11-11T19:31:40.950691800Z" />
+  </component>
+</project>
```
Comment (rt4914) [inline .idea/deploymentTargetDropDown.xml]: We don't need any change to `.idea` file, this must have been done automatically from android studio please do not add this file to your commit. 
Note: Don't delete this file, we need this file we just don't need the changes this file contains.

File: .idea/shelf/Uncommitted_changes_before_Update_at_12-11-2021_16_31_[Default_Changelist]/shelved.patch [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
++  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+     <output url="file://$PROJECT_DIR$/build/classes" />
+   </component>
+   <component name="ProjectType">
+     <option name="id" value="Android" />
+   </component>
+-</project>
++</project>
+\ No newline at end of file
```
Comment (rt4914) [inline .idea/shelf/Uncommitted_changes_before_Update_at_12-11-2021_16_31_[Default_Changelist]/shelved.patch]: We don't need any change to `.idea` file, this must have been done automatically from android studio please do not add this file to your commit. 
Note: Don't delete this file, we need this file we just don't need the changes this file contains.

File: .idea/shelf/Uncommitted_changes_before_Update_at_12-11-2021_16_31__Default_Changelist_.xml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<changelist name="Uncommitted_changes_before_Update_at_12-11-2021_16_31_[Default_Changelist]" date="1636714895106" recycled="true" deleted="true">
+  <option name="PATH" value="$PROJECT_DIR$/.idea/shelf/Uncommitted_changes_before_Update_at_12-11-2021_16_31_[Default_Changelist]/shelved.patch" />
+  <option name="DESCRIPTION" value="Uncommitted changes before Update at 12-11-2021 16:31 [Default Changelist]" />
+</changelist>
```
Comment (rt4914) [inline .idea/shelf/Uncommitted_changes_before_Update_at_12-11-2021_16_31__Default_Changelist_.xml]: We don't need any change to `.idea` file, this must have been done automatically from android studio please do not add this file to your commit. 
Note: Don't delete this file, we need this file we just don't need the changes this file contains.

File: .idea/shelf/Uncommitted_changes_before_Update_at_29-10-2021_00_50_[Default_Changelist]/shelved.patch [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
++            <type value="VIRTUAL_DEVICE_PATH" />
++            <value value="C:\Users\Jishnu Goyal\.android\avd\Tablet_Pixel_C_API_30.avd" />
++          </Key>
++        </deviceKey>
++      </Target>
++    </targetSelectedWithDropDown>
++    <timeTargetWasSelectedWithDropDown value="2021-10-28T19:15:52.715292100Z" />
++  </component>
++</project>
+\ No newline at end of file
```
Comment (rt4914) [inline .idea/shelf/Uncommitted_changes_before_Update_at_29-10-2021_00_50_[Default_Changelist]/shelved.patch]: We don't need any change to `.idea` file, this must have been done automatically from android studio please do not add this file to your commit. 
Note: Don't delete this file, we need this file we just don't need the changes this file contains.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@JishnuGoyal There are unexpected changes. Please undo them.

**Comment by JishnuGoyal** [comment]
I'm sorry, I usually try to make sure to not push changes from that directory. 

PTAL @rt4914 , thanks for being so patient with me :)

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**